src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,152 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size = 0;
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> lastNode = tail;
+        Node<T> newNode = new Node<>(lastNode, value, null);
+        tail = newNode;
+        if (lastNode == null) {
+            head = newNode;
+        } else {
+            lastNode.last = newNode;
+        }
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("Incorrect index");
+        }
+        if (index == size) {
+            add(value);
+        } else {
+            linkBefore(value, nodeIndex(index));
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (int i = 0; i < list.size(); i++) {
+            T value = list.get(i);
+            add(value);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        isIndexExist(index);
+        return nodeIndex(index).element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        isIndexExist(index);
+        Node<T> currentIndex = nodeIndex(index);
+        T oldVal = currentIndex.element;
+        currentIndex.element = value;
+        return oldVal;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        isIndexExist(index);
+        return position(nodeIndex(index));
     }
 
     @Override
     public boolean remove(T object) {
+        for (Node<T> newNode = head; newNode != null; newNode = newNode.last) {
+            if (newNode.element == object
+                    || (newNode.element != null && newNode.element.equals(object))) {
+                position(newNode);
+                return true;
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void isIndexExist(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Element can't be found!");
+        }
+    }
+
+    Node<T> nodeIndex(int index) {
+        Node<T> newNodeHead = head;
+        Node<T> newNodeTail = tail;
+        if (index < (size >> 1)) {
+            for (int i = 0; i < index; i++) {
+                newNodeHead = newNodeHead.last;
+            }
+        } else {
+            for (int i = size - 1; i > index; i--) {
+                newNodeTail = newNodeTail.first;
+            }
+        }
+        return index < (size >> 1) ? newNodeHead : newNodeTail;
+    }
+
+    void linkBefore(T item, Node<T> current) {
```

Comment:
what about access modifier? 
```suggestion
   private void linkBefore(T item, Node<T> current) {
```